src/vcpkg/registries.cpp

```diff
@@ -444,12 +446,23 @@ namespace
 
     struct FilesystemRegistryEntry final : RegistryEntry
     {
-        explicit FilesystemRegistryEntry(std::string&& port_name) : port_name(port_name) { }
+        explicit FilesystemRegistryEntry(const ReadOnlyFilesystem& fs,
+                                         StringView port_name,
+                                         std::vector<FilesystemVersionDbEntry>&& version_entries)
+            : fs(fs), port_name(port_name.data(), port_name.size())
+        {
+            for (auto&& version_entry : version_entries)
+            {
+                port_versions.push_back(std::move(version_entry.version.version));
+                version_paths.push_back(std::move(version_entry.p));
+            }
```

Comment:
This was moved from line 855 to make `FilesystemRegistryEntry` actually be usable when the constructor completes.